services/user/InviteSys/src/InviteSys.cpp

```diff
@@ -0,0 +1,328 @@
+#include <services/system/AccountSys.hpp>
+#include <services/system/AuthAnySys.hpp>
+#include <services/system/AuthEcSys.hpp>
+#include <services/system/ProxySys.hpp>
+#include <services/system/TransactionSys.hpp>
+#include <services/system/commonErrors.hpp>
+#include <services/user/AuthInviteSys.hpp>
+#include <services/user/InviteSys.hpp>
+#include <services/user/NftSys.hpp>
+#include <services/user/TokenSys.hpp>
+
+#include <psibase/Bitset.hpp>
+#include <vector>
+
+using namespace psibase;
+using namespace UserService;
+using namespace UserService::Invite;
+using namespace UserService::Errors;
+using namespace SystemService;
+using psio::const_view;
+using std::begin;
+using std::end;
+using std::move;
+using std::optional;
+using std::string;
+using std::vector;
+
+InviteSys::InviteSys(psio::shared_view_ptr<psibase::Action> action)
+{
+   MethodNumber m{action->method()->value().get()};
+   if (m != MethodNumber{"init"})
+   {
+      auto initRecord = Tables().open<InitTable>().get(SingletonKey{});
+      check(initRecord.has_value(), UserService::Errors::uninitialized);
+   }
+}
+
+void InviteSys::init()
+{
+   auto initTable = Tables().open<InitTable>();
+   auto init      = (initTable.get(SingletonKey{}));
+   check(not init.has_value(), alreadyInit);
+   initTable.put(InitializedRecord{});
+
+   // Register serveSys handler
+   to<SystemService::ProxySys>().registerServer(InviteSys::service);
+
+   // Configure manual debit for self on Token and NFT
+   auto manualDebit = psibase::NamedBit{"manualDebit"};
+   to<TokenSys>().setUserConf(manualDebit, true);
+   to<NftSys>().setUserConf(manualDebit, true);
+
+   // Create the invite payer account and set its auth contract
+   to<AccountSys>().newAccount(payerAccount, AuthInviteSys::service, true);
+
+   // Enable the invite system in AccountSys
+   to<AccountSys>().setCreator(service);
```

Comment:
The way the permission system for AccountSys is set up, calling `setCreator` doesn't allow this invite to create accounts. It's already allowed to do that because of its name. Calling `setCreator` here serves to prevent non-system accounts from creating new accounts. If we get rid of the name based permissions, then we would not be allowed to call `setCreator` here either.

Reply:
Got rid of this whole setCreator system and name-based permissions.